doc/development_guide/contributor_guide/index.rst

```diff
@@ -0,0 +1,12 @@
+Contributing to pylint
+======================
+
+The contributor guide will help you if you want to contribute to pylint itself.
+
+.. toctree::
+   :maxdepth: 2
+   :titlesonly:
+
+   contribute
+   tests/index
+   release
```

Comment:
I would also add the `Profiling` stuff here. Why would anybody need that beside people that want to contribute to `pylint`? One less header in the top level 🎉